Linker errors when building with Xcode 12 and CocoaPods — building for iOS Simulator, but linking in dylib built for iOS

Xcode 12 raises linker errors about architectures when building the navigation SDK using CocoaPods. These warnings also cause CocoaPods to reject any attempt to push the MapboxCoreNavigation or MapboxNavigation pod to CocoaPods trunk.

## Problem details

To validate the fix in #2662 for the CocoaPods linter error in https://github.com/mapbox/mapbox-navigation-ios/pull/2646#issuecomment-699028278, I tried pushing to a throwaway pod on CocoaPods trunk and ran into errors doing so with the combination of CocoaPods 1.10.0.rc.1 and Xcode 12:

```
$ pod trunk push test-1ec5-MapboxCoreNavigation.podspec 
Updating spec repo `trunk`
Validating podspec
 -> test-1ec5-MapboxCoreNavigation (1.0.0)
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - NOTE  | xcodebuild:  note: Using new build system
    - NOTE  | xcodebuild:  note: Building targets in parallel
    - NOTE  | xcodebuild:  note: Using codesigning identity override: -
    - NOTE  | [iOS] xcodebuild:  note: Planning build
    - NOTE  | [iOS] xcodebuild:  note: Constructing build description
    - NOTE  | [iOS] xcodebuild:  warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99. (in target 'Polyline' from project 'Pods')
    - NOTE  | xcodebuild:  clang: error: linker command failed with exit code 1 (use -v to see invocation)
    - NOTE  | [iOS] xcodebuild:  warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99. (in target 'MapboxMobileEvents' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  warning: Skipping code signing because the target does not have an Info.plist file and one is not being generated automatically. (in target 'App' from project 'App')

[!] The spec did not pass validation, due to 1 error.
```

The verbose output contains additional details about the linker errors:

```
    Ld /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation.build/Objects-normal/arm64/Binary/MapboxCoreNavigation normal arm64 (in target 'test-1ec5-MapboxCoreNavigation' from project 'Pods')
        cd /var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target arm64-apple-ios10.0-simulator -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.0.sdk -L/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/MapboxDirections -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/MapboxMobileEvents -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/Polyline -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/Turf -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxAccounts -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxCommon -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxNavigationNative -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/XCFrameworkIntermediates/MapboxCommon -filelist /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation.build/Objects-normal/arm64/MapboxCoreNavigation.LinkFileList -install_name @rpath/MapboxCoreNavigation.framework/MapboxCoreNavigation -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation.build/Objects-normal/arm64/MapboxCoreNavigation_lto.o -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation.build/Objects-normal/arm64/MapboxCoreNavigation.swiftmodule -framework MapboxAccounts -framework MapboxCommon -framework MapboxNavigationNative -framework Foundation -framework MapboxDirections -framework MapboxMobileEvents -framework Turf -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation.build/Objects-normal/arm64/MapboxCoreNavigation_dependency_info.dat -o /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation.build/Objects-normal/arm64/Binary/MapboxCoreNavigation
    ld: building for iOS Simulator, but linking in dylib built for iOS, file '/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxAccounts/MapboxAccounts.framework/MapboxAccounts' for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

```
    Ld /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/arm64/Binary/App normal arm64 (in target 'App' from project 'App')
        cd /var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target arm64-apple-ios10.0-simulator -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.0.sdk -L/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/MapboxDirections -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/MapboxMobileEvents -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/Polyline -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/Turf -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxAccounts -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxCommon -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxNavigationNative -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/XCFrameworkIntermediates/MapboxCommon -filelist /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/arm64/App.LinkFileList -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/arm64/App_lto.o -Xlinker -objc_abi_version -Xlinker 2 -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/arm64/App.swiftmodule -lc++ -lz -framework CoreLocation -framework MapboxAccounts -framework MapboxCommon -framework MapboxCoreNavigation -framework MapboxDirections -framework MapboxMobileEvents -framework MapboxNavigationNative -framework Polyline -framework Turf -Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/App.app-Simulated.xcent -framework Foundation -framework Pods_App -Xlinker -dependency_info -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/arm64/App_dependency_info.dat -o /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/arm64/Binary/App
    ld: framework not found MapboxCoreNavigation
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

```
    Ld /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/x86_64/Binary/App normal x86_64 (in target 'App' from project 'App')
        cd /var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target x86_64-apple-ios10.0-simulator -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.0.sdk -L/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/MapboxDirections -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/MapboxMobileEvents -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/Polyline -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/Turf -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxAccounts -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxCommon -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxNavigationNative -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/XCFrameworkIntermediates/MapboxCommon -filelist /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/x86_64/App.LinkFileList -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/x86_64/App_lto.o -Xlinker -objc_abi_version -Xlinker 2 -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/x86_64/App.swiftmodule -lc++ -lz -framework CoreLocation -framework MapboxAccounts -framework MapboxCommon -framework MapboxCoreNavigation -framework MapboxDirections -framework MapboxMobileEvents -framework MapboxNavigationNative -framework Polyline -framework Turf -Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/App.app-Simulated.xcent -framework Foundation -framework Pods_App -Xlinker -dependency_info -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/x86_64/App_dependency_info.dat -o /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/x86_64/Binary/App
    ld: framework not found MapboxCoreNavigation
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

```
    Ld /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/i386/Binary/App normal i386 (in target 'App' from project 'App')
        cd /var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target i386-apple-ios10.0-simulator -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.0.sdk -L/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/MapboxDirections -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/MapboxMobileEvents -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/Polyline -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/Turf -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxAccounts -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxCommon -F/var/folders/cl/8cgpg37n5cl78pjkbpgml94w0000gn/T/CocoaPods-Lint-20200929-14149-gxhgse-test-1ec5-MapboxCoreNavigation/Pods/MapboxNavigationNative -F/Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Products/Release-iphonesimulator/XCFrameworkIntermediates/MapboxCommon -filelist /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/i386/App.LinkFileList -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/i386/App_lto.o -Xlinker -objc_abi_version -Xlinker 2 -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/i386/App.swiftmodule -lc++ -lz -framework CoreLocation -framework MapboxAccounts -framework MapboxCommon -framework MapboxCoreNavigation -framework MapboxDirections -framework MapboxMobileEvents -framework MapboxNavigationNative -framework Polyline -framework Turf -Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/App.app-Simulated.xcent -framework Foundation -framework Pods_App -Xlinker -dependency_info -Xlinker /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/i386/App_dependency_info.dat -o /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/i386/Binary/App
    ld: framework not found MapboxCoreNavigation
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

```
    ** BUILD FAILED **
    
    
    The following build commands failed:
    	Ld /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/test-1ec5-MapboxCoreNavigation.build/Objects-normal/arm64/Binary/MapboxCoreNavigation normal arm64
    	Ld /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/arm64/Binary/App normal arm64
    	Ld /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/x86_64/Binary/App normal x86_64
    	Ld /Users/mxn/Library/Developer/Xcode/DerivedData/App-benotumsmncznfanmwdstljxbjmd/Build/Intermediates.noindex/App.build/Release-iphonesimulator/App.build/Objects-normal/i386/Binary/App normal i386
    (4 failures)
```

## Workarounds

As a temporary workaround, I `xcode-select`ed Xcode 11.7 and tried again, this time successfully: CocoaPods/Specs@23ef62369f96659bc1c80a64ca3615509eabb329.

```
$ pod trunk push test-1ec5-MapboxCoreNavigation.podspec --allow-warnings
Updating spec repo `trunk`
Validating podspec
 -> test-1ec5-MapboxCoreNavigation (1.0.0)
    - NOTE  | xcodebuild:  note: Using new build system
    - NOTE  | xcodebuild:  note: Building targets in parallel
    - NOTE  | [iOS] xcodebuild:  note: Planning build
    - NOTE  | [iOS] xcodebuild:  note: Constructing build description
    - NOTE  | [iOS] xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'Polyline' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'MapboxMobileEvents' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'Turf' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'MapboxDirections' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'test-1ec5-MapboxCoreNavigation' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'Pods-App' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'App' from project 'App')
    - NOTE  | [iOS] xcodebuild:  warning: Skipping code signing because the target does not have an Info.plist file and one is not being generated automatically. (in target 'App' from project 'App')

Updating spec repo `trunk`

--------------------------------------------------------------------------------
 🎉  Congrats

 🚀  test-1ec5-MapboxCoreNavigation (1.0.0) successfully published
 📅  September 29th, 18:13
 🌎  https://cocoapods.org/pods/test-1ec5-MapboxCoreNavigation
 👍  Tell your friends!
--------------------------------------------------------------------------------
```

However, this workaround only makes it possible to push the pod to CocoaPods trunk; application developers using Xcode  12 are still affected by the linker errors. A more robust workaround requires modifying the podspecs to exclude Apple Silicon from the valid architectures: https://github.com/CocoaPods/CocoaPods/issues/10065#issuecomment-694266259 https://github.com/CocoaPods/CocoaPods/issues/10104#issuecomment-700918704.

## Solution

A more long-term solution will require rebuilding MapboxAccounts with all the architectures required by Xcode 12. In the meantime, we should apply the workaround in https://github.com/CocoaPods/CocoaPods/issues/10065#issuecomment-694266259 https://github.com/CocoaPods/CocoaPods/issues/10104#issuecomment-700918704 to the podspecs in this repository.

/cc @mapbox/navigation-ios @mapbox/core-sdk

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -54,7 +54,8 @@ step-library:
       run:
         name: Update Carthage version
         command: |
-          brew update && brew upgrade carthage
+          curl -OL "https://github.com/Carthage/Carthage/releases/download/0.36.0/Carthage.pkg"
+          sudo installer -pkg Carthage.pkg -target /
 
   - &verify-missing-localizable-strings
       run:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -259,6 +259,17 @@ workflows:
           test: false
           device: "iPhone 7 Plus"
       - pod-job:
+          name: "Xcode_12.0_iOS_14.0_CP_install"
+          update: false
+          xcode: "12.0.0"
+          iOS: "14.0"
+      - pod-job:
+          name: "Xcode_12.0_iOS_14.0_CP_update"
+          update: true
+          xcode: "12.0.0"
+          iOS: "14.0"
+          lint: true
+      - pod-job:
           name: "Xcode_11.4.1_iOS_12.2_CP_install"
           update: false
           xcode: "11.4.1"

--- a/MapboxCoreNavigation.podspec
+++ b/MapboxCoreNavigation.podspec
@@ -48,4 +48,13 @@ Pod::Spec.new do |s|
 
   s.swift_version = "5.0"
 
+  # https://github.com/mapbox/mapbox-navigation-ios/issues/2665
+  s.user_target_xcconfig = {
+    'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => '$(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))',
+    'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200' => 'arm64 arm64e armv7 armv7s armv6 armv8'
+  }
+  s.pod_target_xcconfig = {
+    'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => '$(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))',
+    'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200' => 'arm64 arm64e armv7 armv7s armv6 armv8'
+  }
 end

--- a/MapboxNavigation-Documentation.podspec
+++ b/MapboxNavigation-Documentation.podspec
@@ -57,4 +57,13 @@ Pod::Spec.new do |s|
 
   s.swift_version = "5.0"
 
+  # https://github.com/mapbox/mapbox-navigation-ios/issues/2665
+  s.user_target_xcconfig = {
+    'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => '$(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))',
+    'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200' => 'arm64 arm64e armv7 armv7s armv6 armv8'
+  }
+  s.pod_target_xcconfig = {
+    'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => '$(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))',
+    'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200' => 'arm64 arm64e armv7 armv7s armv6 armv8'
+  }
 end

--- a/MapboxNavigation.podspec
+++ b/MapboxNavigation.podspec
@@ -51,4 +51,9 @@ Pod::Spec.new do |s|
 
   s.swift_version = "5.0"
 
+  # https://github.com/mapbox/mapbox-navigation-ios/issues/2665
+  s.pod_target_xcconfig = {
+    'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => '$(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))',
+    'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200' => 'arm64 arm64e armv7 armv7s armv6 armv8'
+  }
 end

